compose/foundation/foundation/src/skikoTest/kotlin/androidx/compose/foundation/text/selection/CommonSelectionTest.kt

```diff
@@ -0,0 +1,177 @@
+        }
+    }
+
+    @OptIn(ExperimentalTestApi::class)
+    fun selectTextStart() {
+        textFieldSemanticInteraction("line 1\nline 2\nline 3\nline 4\nline 5") { node, state ->
+            node.performKeyInput {
+                pressKey(Key.DirectionRight)
+                pressKey(Key.DirectionDown)
+            }
+
+            waitAndCheck {
+                assertThat(state.value.selection).isEqualTo(TextRange(8, 8))
+            }
+
+            node.performKeyInput { this.selectTextStart() }
+
+            waitAndCheck { assertThat(state.value.selection).isEqualTo(TextRange(8, 0)) }
+        }
+    }
+
+    @OptIn(ExperimentalTestApi::class)
+    fun selectTextEnd() {
+        textFieldSemanticInteraction("line 1\nline 2\nline 3\nline 4\nline 5") { node, state ->
+            node.performKeyInput {
+                pressKey(Key.DirectionRight)
+                pressKey(Key.DirectionDown)
+            }
+                waitAndCheck {
+                    assertThat(state.value.selection).isEqualTo(TextRange(8, 8))
+                }
+                node.performKeyInput { this.selectTextEnd() }
+                waitAndCheck {
+                    assertThat(state.value.selection).isEqualTo(TextRange(8, 34))
+                }
+        }
+    }
+    @OptIn(ExperimentalTestApi::class)
+    fun selectLineEnd() {
+        textFieldSemanticInteraction("line 1\nline 2\nline 3\nline 4\nline 5") { node, state ->
+            node.performKeyInput {
+                pressKey(Key.DirectionRight)
+                pressKey(Key.DirectionDown)
+            }
+            waitAndCheck {
+                assertThat(state.value.selection).isEqualTo(TextRange(8, 8))
+            }
+                node.performKeyInput { this.selectLineEnd() }
+                waitAndCheck {
+                    assertThat(state.value.selection).isEqualTo(TextRange(8, 13))
+                }
+        }
+    }
+
+    @OptIn(ExperimentalTestApi::class)
+    fun deleteAll() {
+        textFieldSemanticInteraction("") { node, state ->
+            node.performKeyInput { this.deleteAll() }
+            waitAndCheck { assertThat(state.value.text).isEqualTo("") }
+        }
+    }
+
+    @OptIn(ExperimentalTestApi::class)
+    fun selectAll() {
+        textFieldSemanticInteraction("Select this text") { node, state ->
+            node.performKeyInput { this.selectAll() }
+            waitAndCheck {
+                assertThat(state.value.selection).isEqualTo(TextRange(0, 16))
+            }
+            node.performKeyInput { keyDown(Key.Delete) }
+            waitAndCheck {
+                assertThat(state.value.selection).isEqualTo(TextRange(0, 0))
+                assertThat(state.value.text).isEqualTo("")
+            }
+        }
+    }
+
+    abstract fun selectLineStartTest()
+    abstract fun selectTextStartTest()
+    abstract fun selectTextEndTest()
+    abstract fun selectLineEndTest()
+    abstract fun deleteAllTest()
+    abstract fun selectAllTest()
```

Comment:
Can you explain why these are needed? Will there be a subclass that does something different from DesktopSelectionTests?